warp fails to compile without the websocket feature

The websocket functionality is behind a default feature so that it can be disabled. However, disabling the feature results in the crate failing to compile with the following error:
```
failed to resolve: maybe a missing crate `ws`?
   --> src/reject.rs:253:37
    |
253 |     MissingConnectionUpgrade(crate::ws::MissingConnectionUpgrade),
    |                                     ^^ maybe a missing crate `ws`?

error: aborting due to previous error
```

The problematic code is behind a macro, so I don't immediately see a way to fix it. The macro was introduced in #311, but the issue appeared to still exist in the commit beforehand, albeit erroring on a different line:
```
error[E0433]: failed to resolve: maybe a missing crate `ws`?
   --> src/reject.rs:400:35
    |
400 |                     e.is::<crate::ws::MissingConnectionUpgrade> () {
    |                                   ^^ maybe a missing crate `ws`?

error: aborting due to previous error
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -35,7 +35,7 @@ jobs:
 
     strategy:
       matrix:
-        build: [stable, beta, nightly, tls]
+        build: [stable, beta, nightly, tls, no-default-features]
 
         include:
           - build: beta
@@ -45,6 +45,9 @@ jobs:
             benches: true
           - build: tls
             features: "--features tls"
+          - build: no-default-features
+            features: "--no-default-features"
+
 
     steps:
       - name: Checkout

--- a/README.md
+++ b/README.md
@@ -1,7 +1,6 @@
 # warp
 
 [![GHA Build Status](https://github.com/seanmonstar/warp/workflows/CI/badge.svg)](https://github.com/seanmonstar/warp/actions?query=workflow%3ACI)
-[![Travis Build Status](https://travis-ci.org/seanmonstar/warp.svg?branch=master)](https://travis-ci.org/seanmonstar/warp)
 [![MIT licensed](https://img.shields.io/badge/license-MIT-blue.svg)](./LICENSE)
 [![crates.io](https://img.shields.io/crates/v/warp.svg)](https://crates.io/crates/warp)
 [![Released API docs](https://docs.rs/warp/badge.svg)](https://docs.rs/warp)

--- a/src/reject.rs
+++ b/src/reject.rs
@@ -190,9 +190,10 @@ enum Rejections {
 }
 
 macro_rules! enum_known {
-    ($($var:ident($ty:path),)+) => (
+     ($($(#[$attr:meta])* $var:ident($ty:path),)+) => (
         pub(crate) enum Known {
             $(
+            $(#[$attr])*
             $var($ty),
             )+
         }
@@ -201,6 +202,7 @@ macro_rules! enum_known {
             fn inner_as_any(&self) -> &dyn Any {
                 match *self {
                     $(
+                    $(#[$attr])*
                     Known::$var(ref t) => t,
                     )+
                 }
@@ -211,6 +213,7 @@ macro_rules! enum_known {
             fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
                 match *self {
                     $(
+                    $(#[$attr])*
                     Known::$var(ref t) => t.fmt(f),
                     )+
                 }
@@ -221,6 +224,7 @@ macro_rules! enum_known {
             fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
                 match *self {
                     $(
+                    $(#[$attr])*
                     Known::$var(ref t) => t.fmt(f),
                     )+
                 }
@@ -229,6 +233,7 @@ macro_rules! enum_known {
 
         $(
         #[doc(hidden)]
+        $(#[$attr])*
         impl From<$ty> for Known {
             fn from(ty: $ty) -> Known {
                 Known::$var(ty)
@@ -250,6 +255,7 @@ enum_known! {
     BodyReadError(crate::body::BodyReadError),
     BodyDeserializeError(crate::body::BodyDeserializeError),
     CorsForbidden(crate::cors::CorsForbidden),
+    #[cfg(feature = "websocket")]
     MissingConnectionUpgrade(crate::ws::MissingConnectionUpgrade),
     MissingExtension(crate::ext::MissingExtension),
     ReplyHttpError(crate::reply::ReplyHttpError),
@@ -378,8 +384,9 @@ impl Rejections {
                 | Known::MissingCookie(_)
                 | Known::InvalidQuery(_)
                 | Known::BodyReadError(_)
-                | Known::BodyDeserializeError(_)
-                | Known::MissingConnectionUpgrade(_) => StatusCode::BAD_REQUEST,
+                | Known::BodyDeserializeError(_) => StatusCode::BAD_REQUEST,
+                #[cfg(feature = "websocket")]
+                Known::MissingConnectionUpgrade(_) => StatusCode::BAD_REQUEST,
                 Known::LengthRequired(_) => StatusCode::LENGTH_REQUIRED,
                 Known::PayloadTooLarge(_) => StatusCode::PAYLOAD_TOO_LARGE,
                 Known::UnsupportedMediaType(_) => StatusCode::UNSUPPORTED_MEDIA_TYPE,

--- a/src/test.rs
+++ b/src/test.rs
@@ -86,6 +86,7 @@ use std::future::Future;
 use std::net::SocketAddr;
 #[cfg(feature = "websocket")]
 use std::pin::Pin;
+#[cfg(feature = "websocket")]
 use std::task::{self, Poll};
 
 use bytes::Bytes;

